shared/reducers/tracker.js

```diff
@@ -1,842 +1,399 @@
 // @flow
-import * as CommonConstants from '../constants/common'
```

Comment:
this has changed enough you likely want to look at it on its own. 
key takeaways:
we have 2 helpers updateUserState and updateNonUserState
and now we have one big switch.

one thing you'll notice is i've wrapped most case statements in curlies as if we do the form
```
case blah: 
  const {foo} = action.payload
return {...somethign}
```

foo now exist throughout the switch statement. This causes 2 problems.
1. if you need a var named the same thing in another switch, it won't let you cause its already defined
2. flow/eslint get confused and think you can access this variable. I had a closure which referred to username (which a lot of things pull out) but really it was undefined.

The real fix is to actually ditch the switch statement (coming soon) so we have totally sep. closures with no leakeage. In this PR (since its huge) i'm not touching this switch that much so i'm just wrapping it in curlies
/shrug